### PR TITLE
Switch synchronize action

### LINKED ISSUE #2189: Add "remove excess" to /switches/{switch_id}/synchronize
Scope:
1. Add "Remove excess=(true/false)" optional parameter to `/switches/{switch_id}/synchronize`. Optional, `False` by default
2. Implement excess flow removal if flag above set to `True` as a part of synchronize operation.
System should be able to remove both excess rules and meters from the switch

### PR DESCRIPTION
**Pull request by IlyaMoiseev**
Closes #2189 
Depends on #2289

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2215)

### CODE REVIEW

File: services/src/northbound/src/main/java/org/openkilda/northbound/controller/SwitchController.java [hunk 1]
```diff
@@ -270,7 +271,21 @@
...
+    /**
+     * Synchronize (install) missing flows that should be on the switch but exist only in neo4j.
+     * Optionally removes excess rules from the switch.
+     *
+     * @return the synchronization result.
+     */
+    @ApiOperation(value = "Synchronize rules on the switch", response = RulesSyncResult.class)
+    @PostMapping(path = "/{switch_id}/synchronize")
+    @ResponseStatus(HttpStatus.OK)
+    public CompletableFuture<RulesSyncResult> syncRules(@PathVariable(name = "switch_id") SwitchId switchId,
```
Comment (rtretyak) [inline services/src/northbound/src/main/java/org/openkilda/northbound/controller/SwitchController.java]: Why we return RulesSyncResult? It should return sth similar to `SwitchValidationResult` but about sync

File: services/wfm/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SyncRulesService.java [hunk 2]
```diff
@@ -27,6 +27,8 @@
 
     void handleInstallRulesResponse(String key);
 
+    void handleRemoveRulesResponse(String key);
```
Comment (dpoltavets) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SyncRulesService.java]: This method is not used anywhere.

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/BatchFlowInstallForSwitchManagerRequest.java [hunk 3]
```diff
@@ -26,15 +26,15 @@
 
 @Value
 @Builder
-public class BatchInstallForSwitchManagerRequest extends CommandData {
+public class BatchFlowInstallForSwitchManagerRequest extends CommandData {
```
Comment (dpoltavets) [inline services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/BatchFlowInstallForSwitchManagerRequest.java]: I would suggest to name `BatchInstallFlowForSwitchManagerRequest` here to match the overall style.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SyncRulesService.java (hunk 2 above)
Comment (IlyaMoiseev) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/switchmanager/service/SyncRulesService.java]: Fixed

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/BatchFlowInstallForSwitchManagerRequest.java (hunk 3 above)
Comment (IlyaMoiseev) [reply services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/BatchFlowInstallForSwitchManagerRequest.java]: Yep, rename reverted

File: services/src/northbound/src/main/java/org/openkilda/northbound/controller/SwitchController.java (hunk 1 above)
Comment (IlyaMoiseev) [reply services/src/northbound/src/main/java/org/openkilda/northbound/controller/SwitchController.java]: fixed

**Pull request closed without merging**